pkg/utils/nodeutils.go

```diff
@@ -115,6 +121,79 @@ func IsMcfgPoolUsingKC(pool *mcfgv1.MachineConfigPool) (bool, string, error) {
 	return true, currentKCMC, nil
 }
 
+func AreKubeletConfigsRendered(pool *mcfgv1.MachineConfigPool, client runtimeclient.Client) (bool, error) {
+	// find out if pool is using a custom kubelet config
+	isUsingKC, currentKCMCName, err := IsMcfgPoolUsingKC(pool)
+	if err != nil {
+		return false, fmt.Errorf("failed to check if pool %s is using a custom kubelet config: %w", pool.Name, err)
+	}
+	if !isUsingKC || currentKCMCName == "" {
+		return true, nil
+	}
+	// if the pool is using a custom kubelet config, check if the kubelet config is rendered
+	kcmcfg := &mcfgv1.MachineConfig{}
+	err = client.Get(context.TODO(), types.NamespacedName{Name: currentKCMCName, Namespace: pool.Namespace}, kcmcfg)
+	if err != nil {
+		return false, fmt.Errorf("failed to get machine config %s: %w", currentKCMCName, err)
+	}
+
+	kc, err := GetKCFromMC(kcmcfg, client)
+	if err != nil {
+		return false, fmt.Errorf("failed to get kubelet config from machine config %s: %w", currentKCMCName, err)
+	}
+
+	var obj interface{}
+	err = json.Unmarshal(kcmcfg.Spec.Config.Raw, &obj)
+	if err != nil {
+		return false, fmt.Errorf("failed to unmarshal machine config %s: %w", currentKCMCName, err)
+	}
+
+	encodedKC, err := jsonpath.Get("storage.files[0].contents.source", obj)
+	if err != nil {
+		return false, fmt.Errorf("failed to get encoded kubelet config from machine config %s: %w", currentKCMCName, err)
+	}
+	encodedKCStr := encodedKC.(string)
+	if encodedKCStr == "" {
+		return false, fmt.Errorf("encoded kubeletconfig %s is empty", currentKCMCName)
+	}
+	mcPayloadPrefix := `data:text/plain,`
+	encodedKCStr = strings.TrimPrefix(encodedKCStr, mcPayloadPrefix)
+
+	decodedKC, err := url.PathUnescape(encodedKCStr)
+	if err != nil {
+		return false, fmt.Errorf("failed to decode kubeletconfig %s: %w", currentKCMCName, err)
+	}
+
+	isSubset, diff, err := JSONIsSubset(kc.Spec.KubeletConfig.Raw, []byte(decodedKC))
+	if err != nil {
+		return false, fmt.Errorf("failed to check if kubeletconfig %s is subset of %s: %w", currentKCMCName, decodedKC, err)
+	}
+	if isSubset {
+		return true, nil
+	} else {
+		PrintJSONDiff(diff)
```

Comment:
hmm, on one hand I like printing the diff, on the other hand shouldn't we go through the logger interface rather than printing the diff directly to stdout? If you don't like to add logging to this function, returning the diff up or writing it into a io.Writer or smh would both work better IMO than an utility function printing to stdout..